src/rosdiscover/Issue74.py

```diff
@@ -0,0 +1,102 @@
+import time
+import collections
+import sys
+
+sys.path.append("/interpreter")
+from summary import NodeSummary
+
+
+from loguru import logger
+
+import roswire
+
+f = open("Arch.yml", "w")
+
+# enable logging
+logger.enable('roswire')
+
+
+# first, we create a new ROSWire session
+rsw = roswire.ROSWire()
+
+def toString(l): 
+    s = ''
+    l1 = list(l)
+    if len(l1) == 0: 
+        return '[]'
+    for i in l1:
+        (fmt, topic) = i
+        s = s + "\n  - format: " + fmt + "\n    name: " + topic + "\n" 
+    return s
+
+# we then launch a container for a given robot.
+# once the end of the given context is reached, including the event where an
+# exception occurs, the container will be automatically destroyed.
+image = 'therobotcooperative/turtlebot3'
+sources = ['/opt/ros/kinetic/setup.bash', '/ros_ws/devel/setup.bash']
+environment = {'TURTLEBOT3_MODEL': 'burger'}
+with rsw.launch(image, sources, environment=environment) as system:
+
+    # we then launch ROS inside the container
+    # again, once the end of the context is reached, ROS is killed.
+    with system.roscore() as ros:
+
+        # let's bring up the application
+        ros.roslaunch('turtlebot3_house.launch',
+                      package='turtlebot3_gazebo',
+                      args={'gui': 'false'})
+
+        # we need to wait for the nodes to finish their initialisation
+        time.sleep(30)
+        
+        node_names = list(ros.nodes)
+
+        # get an overview of the system state
+        state = ros.state
+
+        # get topic types
+        topic_to_type = ros.topic_to_type
+
+        # get service formats
+        
+        service_to_format = {}
+        for service_name in state.services:
+            service = ros.services[service_name]
+            service_to_format[service_name] = service.format.name
+        
+
+nodeSummaryDict = {}
```

Comment:
I think I would refactor this a separate function too. That takes a state and returns NodeSummaryDict.

I think also, for this, you could try using mypy typing like we do in the other parts of the code.